### PR TITLE
[REF] mail: make thread actions more generic

### PR DESCRIPTION
**Pull request by seb-odoo**
This will remove the necessity to patch component, and remove
duplciation between discuss and chat window.

Part of task-3265211

### CODE REVIEW

File: addons/mail/static/src/discuss/core/channel_invitation.scss [hunk 1]
```diff
@@ -1,3 +1,7 @@
+.o-discuss-ChannelInvitation {
+    min-height: 0;
+}
```
Comment (alexkuhn) [inline addons/mail/static/src/discuss/core/channel_invitation.scss:3]: There's `o-min-width-0` very soon in master ;)

File: addons/mail/static/src/scss/variables/primary_variables.scss [hunk 2]
```diff
@@ -2,7 +2,7 @@ $o-mail-Avatar-size: 42px !default;
 $o-mail-ChatWindow-width: 360px !default; // same value as CHAT_WINDOW_WIDTH
 $o-mail-Chatter-mobile-gap: 2% !default;
 $o-mail-sidebar-icon-opacity: 0.7 !default;
-$o-mail-Discuss-inspector: 250px !default;
+$o-mail-Discuss-inspector: 300px !default;
```
Comment (alexkuhn) [inline addons/mail/static/src/scss/variables/primary_variables.scss:5]: Note that pinned message panel takes 425px (it uses `flex-basis`, and it has precedence over `width`).
I'm not entirely sure we want to keep different sizes for inspectors. If that's the case, would be nice to define it in action

Comment (seb-odoo) [reply addons/mail/static/src/scss/variables/primary_variables.scss:5]: I was wondering the same but I feel 425 seems too big for others that are more permanently open (eg. member list), but less than 425 is too small for pinned messages.

Comment (alexkuhn) [reply addons/mail/static/src/scss/variables/primary_variables.scss:5]: Could be worth having 2 official sizes for inspector

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

**Comment by seb-odoo** [comment]
@robodoo retry

File: addons/mail/static/src/core/thread_actions.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        /** Optional component that should be displayed in the view when this action is active. */
+        component: action.component,
+        /** Condition to display the component of this action. */
+        get componentCondition() {
+            return this.isActive && this.component && this.condition && !this.popover;
+        },
+        /** Props to pass to the component of this action. */
+        componentProps: action.componentProps,
+        /** Condition to display this action. */
+        get condition() {
```
Comment (oomsveta) [inline addons/mail/static/src/core/thread_actions.js:39]: I would have called it “predicate”

Comment (seb-odoo) [reply addons/mail/static/src/core/thread_actions.js:39]: You can make a follow up PR to rename it if we think it is better.

**Pull request closed without merging**